zcash_primitives/src/transaction/txid.rs

```diff
@@ -347,17 +344,15 @@ impl<A: Authorization> TransactionDigest<A> for TxIdDigester {
 
     fn digest_orchard(
         &self,
-        orchard_bundle: Option<&bundle::Bundle<A::OrchardAuth, Amount, OrchardVanilla>>,
+        orchard_bundle: Option<&OrchardBundle<A::OrchardAuth>>,
     ) -> Self::OrchardDigest {
-        orchard_bundle.map(|b| b.commitment().0)
-    }
-
-    #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
-    fn digest_orchard_zsa(
-        &self,
-        orchard_bundle: Option<&bundle::Bundle<A::OrchardZsaAuth, Amount, OrchardZSA>>,
-    ) -> Self::OrchardDigest {
-        orchard_bundle.map(|b| b.commitment().0)
+        orchard_bundle.map(|b| {
+            match b {
+                OrchardBundle::OrchardVanilla(vanilla_bundle) => vanilla_bundle.commitment().0,
```

Comment:
reduce boilerplate
`OrchardVanilla(b) => b.commitment().0,` (import + rename) (x2)

Reply:
Not sure what we reduce here. Just var name length? Then it's shadowing which is also not very nice

Reply:
as discussed, import `OrchardVanilla`and `vanilla_bundle` -> `v` ..